File: Erettsegi/emelt/2016_may/Otszaz.java
import java.io.*;
import java.nio.file.*;
import java.util.*;
 
public class Otszaz {
	
    public static void main(String[] args) throws IOException {
        var lines = Files.readAllLines(Paths.get("penztar.txt"));
        var vasarlasok = new ArrayList<Vasarlas>();
       
        ArrayList<String> toAdd = new ArrayList<>();
        for(String sor : lines) {
            if(sor.equals("F")) {
                vasarlasok.add(new Vasarlas(toAdd));
                toAdd.clear();
            }else{
                toAdd.add(sor);
            }
        }
		
        System.out.println("Vásárlások száma: " + vasarlasok.size());
        System.out.println("Elsö vásárlásnál vett dolgok száma: " + vasarlasok.get(0).dolgok.size());
		
        Scanner input = new Scanner(System.in);
        System.out.println("Írj be 1 sorszámot");
        int sorszam = input.nextInt();
        System.out.println("Írj be 1 árut");
        String aru = input.next();
        System.out.println("Írj be 1 mennyiséget");
        int dbszam = input.nextInt();
        input.close();
        
        int amount = 0, utolso = 0;
        for(int k = 0; k < vasarlasok.size(); ++k) {
            for(var entries : vasarlasok.get(k).dolgok.entrySet()) {
                if(entries.getKey().equals(aru)) {
                    ++amount;
                    utolso = k;
                    if(amount == 1) {
                        System.out.println("Elõször a " + ++k + ". vásárlásnál vettek " + aru + "-t");
                    }
                }
            }
        }
       
        System.out.println("Utoljára a " + ++utolso + ". vásárlásnál vettek " + aru + "-t");
        System.out.println("Összesen " + amount + "-szor vettek " + aru + "-t");
        System.out.println(dbszam + " db esetén a fizetendõ: " + ertek(dbszam));
        System.out.println("A " + sorszam + ". vásárláskor vásárolt dolgok: " + vasarlasok.get(sorszam - 1).dolgok.toString());
       
        try(PrintWriter output = new PrintWriter("osszeg.txt")){
	        for(int k = 0; k < vasarlasok.size(); ++k) {
	            int printMount = 0;
	            for(var entries : vasarlasok.get(k).dolgok.entrySet()) {
	                printMount += ertek(entries.getValue());
	            }
	            output.println(Integer.toString(k + 1) + ":" + printMount);
	        }
        }
    }
    
    public static int ertek(int dbSzam) {
        if(dbSzam == 1) {
            return 500;
        }else if(dbSzam == 2) {
            return 950;
        }else if(dbSzam == 3) {
            return 1350;
        }
        return 1350 + (500 * (dbSzam - 1));
    }
    
    static class Vasarlas{
        HashMap<String, Integer> dolgok = new HashMap<>();
        
        public Vasarlas(ArrayList<String> things) {
            for(String th : things) {
                if(!dolgok.containsKey(th)) {
                    dolgok.put(th, Collections.frequency(things, th));
                }
            }
        }
    }
}

File: erettsegi/emelt/2016_may/c#/Otszaz.cs
using System;
using System.Collections.Generic;
using System.IO;

var lines = File.ReadAllLines("penztar.txt");
var vasarlasok = new List<Dictionary<string, int>>();
var itemBuffer = new List<string>();

foreach(var line in lines) {
    if(line == "F") {
        var itemFreqMap = new Dictionary<string, int>();

        foreach(var item in itemBuffer) {
            itemFreqMap.Add(item, itemFreqMap.GetValueOrDefault(item, 0) + 1);
        }

        vasarlasok.Add(itemFreqMap);
        itemBuffer.Clear();
    }else{
        itemBuffer.Add(line);
    }
}

Console.WriteLine($"2. Feladat: Vásárlások száma: {vasarlasok.Count}");
Console.WriteLine($"3. Feladat: Elsö vásárlásnál vett dolgok száma: {vasarlasok[0].Count}");

Console.WriteLine("4. Feladat:");
Console.WriteLine("Írj be 1 sorszámot");
var bekertSorszam = int.Parse(Console.ReadLine());

Console.WriteLine("Írj be 1 árut");
var bekertAru = Console.ReadLine();

Console.WriteLine("Írj be 1 mennyiséget");
var bekertDbszam = int.Parse(Console.ReadLine());

Console.WriteLine("5. Feladat:");

var bekertAruOsszDbSzam = 0;
var bekertAruUtolsoVasarlasIndex = 0;

for(var k = 0; k < vasarlasok.Count; ++k) {
    foreach(var item in vasarlasok[k].Keys) {
        if(item == bekertAru) {
            ++bekertAruOsszDbSzam;
            bekertAruUtolsoVasarlasIndex = k;

            if(bekertAruOsszDbSzam == 1) {
                Console.WriteLine("Először a " + (k + 1) + ". vásárlásnál vettek " + bekertAru + "-t");
            }
        }
    }
}

Console.WriteLine($"Utoljára a {bekertAruUtolsoVasarlasIndex + 1}. vásárlásnál vettek {bekertAru}-t");
Console.WriteLine($"Összesen {bekertAruOsszDbSzam}-szor vettek {bekertAru}-t");
Console.WriteLine($"6. Feladat: {bekertDbszam} db esetén a fizetendő: {ertek(bekertDbszam)}");
Console.WriteLine($"7. Feladat: A {bekertSorszam}. vásárláskor vásárolt dolgok:");

foreach(var (item, db) in vasarlasok[bekertSorszam - 1]) {
    Console.WriteLine($"{item}-ből: {db} db");
}

using var output = new StreamWriter("osszeg.txt");

for(int k = 0; k < vasarlasok.Count; ++k) {
    var printMount = 0;

    foreach(var dbSzam in vasarlasok[k].Values) {
        printMount += ertek(dbSzam);
    }
    output.WriteLine((k + 1).ToString() + ":" + printMount);
}

int ertek(int dbSzam) => dbSzam == 1 ? 500 : dbSzam == 2 ? 950 : 1350 + (400 * (dbSzam - 3));